### DISCUSSION TITLE
Parametrized pages not found

### DISCUSSION THREAD

**Discussion by johanna-einsiedler** [observablehq/framework#1771] [Q&A] [answered]
I am struggling with the implementation of the parametrized pages functionality.

Following [Parameterized routes instructions](https://observablehq.com/framework/params#parameterized-routes), I created a folder called "products" in my src file with a document named [product].md

I further added the following to my config:  
`dynamicPaths: [
    "/products/100736",
    "/products/100739"
  ]`

However, the page doesn't get recognized as a parametrized page, i.e. I get an error "The current product is
RuntimeError: observable is not defined" when calling  "The current product is ${observable.params.product}" within [product].md

Would really appreciate it if anyone could point me towards what went wrong here.

Thanks a lot!

**Comment by mbostock** [comment]
I’m not able to reproduce this error. What version of Framework are you using? Parameterized routes were added in 1.11.0, and the latest release is currently 1.12.0.

To test, I created a new project with the following structure:

```
.
├── observablehq.config.js
├── package.json
├── src
│   ├── index.md
│   └── products
│       └── [product].md
└── yarn.lock
```

In observablehq.config.js:

```js
export default {
  root: "src",
  dynamicPaths: ["/products/100736", "/products/100739"]
};
```

In src/products/[product].md:

```md
The current product is ${observable.params.product}.
```

Running `yarn build` produces the following output:

```
load /index in 58ms
load /products/100736 in 39ms
load /products/100739 in 22ms
built 3 pages in src
bundle /_observablehq/client.js → src/.observablehq/cache/_observablehq/client.js
bundle /_observablehq/runtime.js → src/.observablehq/cache/_observablehq/runtime.js
bundle /_observablehq/stdlib.js → src/.observablehq/cache/_observablehq/stdlib.js
build /_observablehq/theme-air,near-midnight.css → dist/_observablehq/theme-air,near-midnight.3268c726.css
alias /_observablehq/client.js → /_observablehq/client.91eecc2d.js
alias /_observablehq/runtime.js → /_observablehq/runtime.f8298b77.js
alias /_observablehq/stdlib.js → /_observablehq/stdlib.e1df28da.js
build /_observablehq/client.js → dist/_observablehq/client.91eecc2d.js
build /_observablehq/runtime.js → dist/_observablehq/runtime.f8298b77.js
build /_observablehq/stdlib.js → dist/_observablehq/stdlib.e1df28da.js
render /index → dist/index.html
render /products/100736 → dist/products/100736.html
render /products/100739 → dist/products/100739.html

┌ (3 pages)                     Page      Imports        Files
├── products (2 pages)
│   ├── 100736                  2 kB        51 kB        26 kB
│   └── 100739                  2 kB        51 kB        26 kB
└── index                       2 kB        51 kB        26 kB

✨  Done in 1.38s.
```

If you can share more details about your setup or a reproduction, I’d be happy to take another look.

**Comment by johanna-einsiedler** [comment, marked as answer]
Thanks a lot - the version I had been running was too old, I hadn't noticed that this feature has only been introduced that recently. After updating its working! Sorry for bothering you and thanks for the quick answer!
